web/web_test.go

```diff
@@ -708,3 +706,22 @@ func TestMultipleListenAddresses(t *testing.T) {
 		cleanupTestResponse(t, resp)
 	}
 }
+
+// Give some time for the web goroutine to run since we need the server
+// to be up before starting tests.
+func waitForServerReady(t *testing.T, baseURL string, timeout time.Duration) {
```

Comment:
`Healthy` or `Live` would be more appropriate as readiness is related to `/-/ready`

Reply:
alos, you may want to use `require.Eventually()` in here